packages/contracts/src.ts/index.ts

```diff
@@ -1177,46 +1178,33 @@ export class ContractFactory {
         defineReadOnly(this, "signer", signer || null);
     }
 
-    // @TODO: Future; rename to populateTransaction?
-    getDeployTransaction(...args: Array<any>): TransactionRequest {
-        let tx: TransactionRequest = { };
+    getDeployTransactions(...args: Array<any>): Array<TransactionRequest> {
+        let chunks = splitInChunks(Buffer.from(this.bytecode).toString(), 4096);
 
-        // If we have 1 additional argument, we allow transaction overrides
-        if (args.length === this.interface.deploy.inputs.length + 1 && typeof(args[args.length - 1]) === "object") {
-            tx = shallowCopy(args.pop());
-            for (const key in tx) {
-                if (!allowedTransactionKeys[key]) {
-                    throw new Error("unknown transaction override " + key);
-                }
+        const fileCreate: TransactionRequest = {
+            customData: {
+                fileChunk: chunks[0]
             }
-        }
+        };
 
-        // Do not allow these to be overridden in a deployment transaction
-        ["data", "from", "to"].forEach((key) => {
-            if ((<any>tx)[key] == null) { return; }
-            logger.throwError("cannot override " + key, Logger.errors.UNSUPPORTED_OPERATION, { operation: key })
-        });
+        let fileAppends: Array<any> = [];
+        for (let chunk of chunks.slice(1)) {
+            const fileAppend: TransactionRequest = {
+                customData: {
+                    fileChunk: chunk
+                }
+            };
 
-        if (tx.value) {
-            const value = BigNumber.from(tx.value);
-            if (!value.isZero() && !this.interface.deploy.payable) {
-                logger.throwError("non-payable constructor cannot override value", Logger.errors.UNSUPPORTED_OPERATION, {
-                    operation: "overrides.value",
-                    value: tx.value
-                });
-            }
+            fileAppends.push(fileAppend);
         }
 
-        // Make sure the call matches the constructor signature
-        logger.checkArgumentCount(args.length, this.interface.deploy.inputs.length, " in Contract constructor");
-
-        // Set the data to the bytecode + the encoded constructor arguments
-        tx.data = hexlify(concat([
-            this.bytecode,
-            this.interface.encodeDeploy(args)
-        ]));
+        const contractCreate: TransactionRequest = {
+            gasLimit: 300000,
```

Comment:
The gas limit must not be hardcoded here. It must be passed as `transaction override` from the outside